azure/datalake/store/multithread.py

```diff
@@ -469,18 +469,17 @@ def put_chunk(adlfs, src, dst, offset, size, buffersize, blocksize, delimiter=No
             end = offset + size
             miniblock = min(size, blocksize)
             # For empty files there is no need to take the IO hit.
-            if size != 0:
+            if size == 0:
+                # need to explicitly flush the file in order to create it on the server when the file is empty.
+                fout.flush()
```

Comment:
I still think your close should flush and this is useless and counterproductive. Looking at your close implementation, there is a flush:
https://github.com/Azure/azure-data-lake-store-python/blob/master/azure/datalake/store/core.py#L706

If you miss the flush for an empty, I suggest to fix ADLFile instead.